tools/bin/commands/util/openshift_funcs

```diff
@@ -1,11 +1,66 @@
 #!/bin/bash
 
+OC_MIN_VERSION=3.9.0
+
+compare_version_part() {
+    local test=$1
+    local min=$2
+
+    test=$(($test + 0))
+    min=$(($min + 0))
+
+    if [ $test -eq $min ]; then
+        return 0;
+    elif [ $test -gt $min ]; then
+        return 1;
+    else
+        return 2;
+    fi
+}
+
+check_oc_version()
+{
+    set +e
+    local min=${OC_MIN_VERSION}
+    local test=$(oc version | grep oc | tr -d oc\ v | cut -f1 -d "+")
+
+    echo -n "Testing oc version '$test' against required minimum ..... "
+
+    testparts=( ${test//./ } )
+    minparts=( ${min//./ } )
+
+    compare_version_part ${testparts[0]} ${minparts[0]}
+    ret=$?
+    if [ $ret -gt 1 ]; then
+        echo "Error: oc version should be at least $min"
+        exit 1
+    fi
+
+    compare_version_part ${testparts[1]} ${minparts[1]}
+    ret=$?
+    if [ $ret -gt 1 ]; then
+        echo "Error: oc version should be at least $min"
+        exit 1
+    fi
+
+    ret=$(compare_version_part ${testparts[2]} ${minparts[2]})
+    ret=$?
+    if [ $ret -gt 1 ]; then
+        echo "Error: oc version should be at least $min"
+        exit 1
+    fi
+
+    echo "OK"
+    set -e
```

Comment:
If we are using the eval convention for return valus, there is no need to switch on/off -e